stactools_landsat/stactools/landsat/utils.py

```diff
@@ -0,0 +1,141 @@
+import datetime
+
+import dateutil
+import rasterio
+from pystac import Item, Link
+from rasterio import RasterioIOError
+from shapely.geometry import box, mapping, shape
+
+
+def _parse_date(in_date: str) -> datetime.datetime:
+    """
+    Try to parse a date and return it as a datetime object with no timezone
+    """
+    dt = dateutil.parser.parse(in_date)
+    return dt.replace(tzinfo=datetime.timezone.utc)
+
+
+def transform_mtl_to_stac(metadata: dict) -> Item:
+    """
+    Handle USGS MTL as a dict and return a STAC item.
+
+    NOT IMPLEMENTED
+
+    Issues include:
+        - There's no reference to UTM Zone or any other CRS info in the MTL
+        - There's no absolute file path or reference to a URI to find data.
+    """
+    LANDSAT_METADATA = metadata["LANDSAT_METADATA_FILE"]
+    product = LANDSAT_METADATA["PRODUCT_CONTENTS"]
+    projection = LANDSAT_METADATA["PROJECTION_ATTRIBUTES"]
+    image = LANDSAT_METADATA["IMAGE_ATTRIBUTES"]
+    proessing_record = LANDSAT_METADATA["LEVEL2_PROCESSING_RECORD"]
+
+    scene_id = product["LANDSAT_PRODUCT_ID"]
+
+    xmin, xmax = float(projection["CORNER_LL_LON_PRODUCT"]), float(
+        projection["CORNER_UR_LON_PRODUCT"])
+    ymin, ymax = float(projection["CORNER_LL_LAT_PRODUCT"]), float(
+        projection["CORNER_UR_LAT_PRODUCT"])
+    geom = mapping(box(xmin, ymin, xmax, ymax))
+    bounds = shape(geom).bounds
+
+    # Like: "2020-01-01" for date and  "23:08:52.6773140Z" for time
+    dt = _parse_date(f"{image['DATE_ACQUIRED']}T{image['SCENE_CENTER_TIME']}")
+    created = _parse_date(proessing_record["DATE_PRODUCT_GENERATED"])
+
+    item = Item(id=scene_id,
+                geometry=geom,
+                bbox=bounds,
+                datetime=dt,
+                properties={})
+
+    # Common metadata
+    item.common_metadata.created = created
+    item.common_metadata.platform = image["SPACECRAFT_ID"]
+    item.common_metadata.instruments = [
+        i.lower() for i in image["SENSOR_ID"].split("_")
+    ]
+
+    # TODO: implement these three extensions
+    item.ext.enable("eo")
+    item.ext.enable("view")
+    item.ext.enable("projection")
+
+    return item
+
+
+def transform_stac_to_stac(item: Item,
+                           enable_proj: bool = True,
+                           self_link: str = None,
+                           source_link: str = None) -> Item:
+    """
+    Handle a 0.7.0 item and convert it to a 1.0.0.beta2 item.
```

Comment:
This could use some explanation about what the expected input is...are these the STAC items published as part of the Collection 2 data?

Reply:
Yeah Collection 2 includes a STAC collection but it's 0.7.0